src/schema/userExperience.ts

```diff
@@ -267,5 +393,107 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         hits,
       };
     },
+    userExperiences: async (
+      _,
+      params: ExperienceQueryParams,
+      ctx: AuthContext,
+    ) => {
+      const { userId } = ctx;
+
+      const {
+        data,
+        success: passedValidation,
+        error,
+      } = experiences.validation.queryAll.safeParse(params);
+
+      if (!passedValidation) {
+        throw new Error(`Invalid parameters: ${error.message}`);
+      }
+
+      const { status } = data;
+
+      const entries = await queryReadReplica(ctx.con, ({ queryRunner }) =>
+        queryRunner.manager
+          .getRepository(UserExperience)
+          .createQueryBuilder('experience')
+          .where('experience.userId = :userId', { userId })
+          .andWhere('experience.status IN (:...status)', {
+            status,
+          })
+          .addOrderBy('experience.endDate', 'DESC', 'NULLS FIRST')
+          .addOrderBy('experience.startDate', 'DESC')
+          .getMany(),
```

Comment:
Can we apply some kind of limit here, it can be big eg. 100 just to be safe and not query a lot of rows from the database, I understand user will probably not have as many of those.